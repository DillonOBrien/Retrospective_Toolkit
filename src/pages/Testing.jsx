import React from "react";
import { v4 as uuidv4 } from "uuid"; // clear annoying console warning about unique keys
import iconsArr from "../constants/IconsDataStructure";

function getIcons(finalIdx) {
  const icons = [];
  for (let i = 0; i < finalIdx.length; i += 1) {
    const n = finalIdx[i];
    icons.push(iconsArr[n]);
  }
  // eslint-disable-next-line no-console
  // console.log(icons);
  return icons;
}

function getRandom(dieNumber) {
  let rand;
  const loopLength = iconsArr.length * 10;
  const finalIdx = [];
  let count = 0;

  for (let i = 0; i < loopLength; i += 1) {
    rand = Math.floor(Math.random() * iconsArr.length + 1);

    if (count === dieNumber) {
      // eslint-disable-next-line no-console
      console.log(finalIdx);
      break;
    }

    if (!finalIdx.includes(rand)) {
      finalIdx.push(rand);
      count += 1;
    }
  }
  const ret = getIcons(finalIdx);
  return ret;
}

const randomSelected = getRandom(36);
// eslint-disable-next-line no-console
// console.log(randomSelected);

// iterate and renders array.
const Testing = () => (
  <div>
    <h2>Random Selected: </h2>
    <div>
      {randomSelected.map((icon) => (
        <span>
          <img src={icon} alt={icon} />{" "}
        </span>
      ))}
    </div>
    <h1>Testing DS -- Total Items: {iconsArr.length} </h1>
    {iconsArr.map((icon) => (
      <div>
        <li key={uuidv4()}>
          {" "}
          <img src={icon} alt={icon} /> Name: {icon}
        </li>
      </div>
    ))}
  </div>
);

export default Testing;
