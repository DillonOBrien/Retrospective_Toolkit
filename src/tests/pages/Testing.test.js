import React from "react";
import { shallow } from "enzyme";
import Testing from "../../pages/Testing";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Testing />);
});

describe("Pages/Testing", () => {
  // Snapshot for Testing
  test("SNAPSHOT: Testing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
