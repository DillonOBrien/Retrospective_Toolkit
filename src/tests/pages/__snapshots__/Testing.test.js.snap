// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Testing SNAPSHOT: Testing 1`] = `
<div>
  <h2>
    Random Selected: 
    6
     
  </h2>
  <div>
    <span>
      <img
        alt="remove_red_eye-24px.svg"
        key="5a2ac644-bc63-4724-91ba-5fd1e70c1208"
        src="remove_red_eye-24px.svg"
      />
       
    </span>
    <span>
      <img
        alt="bubble_chart-24px.svg"
        key="f9bf3dd6-3599-4391-95d3-406c5ef1c92c"
        src="bubble_chart-24px.svg"
      />
       
    </span>
    <span>
      <img
        alt="toys-24px.svg"
        key="01086100-3c7b-4864-80c2-6833a566c1c3"
        src="toys-24px.svg"
      />
       
    </span>
    <span>
      <img
        alt="straighten-24px.svg"
        key="130197f5-3c10-49c9-b7fe-948538bb6e6f"
        src="straighten-24px.svg"
      />
       
    </span>
    <span>
      <img
        alt="cloud_queue-24px.svg"
        key="4099b9e5-5f62-42ef-b13c-2bf13b719f92"
        src="cloud_queue-24px.svg"
      />
       
    </span>
    <span>
      <img
        alt="music_note-24px.svg"
        key="f3fe8ec9-0d19-4932-9682-deda997249b8"
        src="music_note-24px.svg"
      />
       
    </span>
  </div>
  <h1>
    Testing DS -- Total Items: 
    48
     
  </h1>
  <div>
    <li
      key="e7483e59-1456-475d-acee-cfd6ff14a212"
    >
       
      <img
        alt="alt_route-24px.svg"
        key="8e1733ef-1ab5-4474-b81a-1cf5272300b6"
        src="alt_route-24px.svg"
      />
       Name: 
      alt_route-24px.svg
    </li>
  </div>
  <div>
    <li
      key="b0287ddb-a20b-4424-8f42-ddd1611ed62f"
    >
       
      <img
        alt="brush-24px.svg"
        key="c3f39dca-9629-4423-a242-ab8b2836c9b0"
        src="brush-24px.svg"
      />
       Name: 
      brush-24px.svg
    </li>
  </div>
  <div>
    <li
      key="a7e18f71-13d3-4f90-8106-ed6da7a99a1e"
    >
       
      <img
        alt="directions_run-24px.svg"
        key="5bf86dfd-6273-4788-963c-e8f2f85ec328"
        src="directions_run-24px.svg"
      />
       Name: 
      directions_run-24px.svg
    </li>
  </div>
  <div>
    <li
      key="921853a6-eae3-41ab-b6a6-c166b52f201b"
    >
       
      <img
        alt="directions_walk-24px.svg"
        key="e3e87ce2-2013-4aa5-94b3-85c84368f256"
        src="directions_walk-24px.svg"
      />
       Name: 
      directions_walk-24px.svg
    </li>
  </div>
  <div>
    <li
      key="2485a053-4f54-486c-9a6e-13b9d3f8df78"
    >
       
      <img
        alt="edit-black-18dp.svg"
        key="2b6cee8d-b5e6-4243-9692-9adcf688c013"
        src="edit-black-18dp.svg"
      />
       Name: 
      edit-black-18dp.svg
    </li>
  </div>
  <div>
    <li
      key="6faeb6e4-1483-4f0b-b62e-7f1ed2368096"
    >
       
      <img
        alt="emoji_objects-24px.svg"
        key="ddefe2d7-67f3-46dd-9262-9a21a8b2cd3e"
        src="emoji_objects-24px.svg"
      />
       Name: 
      emoji_objects-24px.svg
    </li>
  </div>
  <div>
    <li
      key="a4fa7d3a-55c9-41f1-b918-1f960a7eaca3"
    >
       
      <img
        alt="grass-24px.svg"
        key="e215ebb3-e25f-4c34-a2d8-9b4c7e073e94"
        src="grass-24px.svg"
      />
       Name: 
      grass-24px.svg
    </li>
  </div>
  <div>
    <li
      key="08c2ca73-8e60-415e-a0cb-5757b587bf56"
    >
       
      <img
        alt="headset-24px.svg"
        key="3a9814f2-db9f-4e5d-b6b0-2d8653ed8ed9"
        src="headset-24px.svg"
      />
       Name: 
      headset-24px.svg
    </li>
  </div>
  <div>
    <li
      key="e3daf809-5a09-426b-a477-e867f1c8b3e1"
    >
       
      <img
        alt="highlight-24px.svg"
        key="25510456-d510-4cb6-958c-1e9d4479c31c"
        src="highlight-24px.svg"
      />
       Name: 
      highlight-24px.svg
    </li>
  </div>
  <div>
    <li
      key="2a533dbf-b314-4663-a9c3-80dc93098334"
    >
       
      <img
        alt="music_note-24px.svg"
        key="8c9353c9-3cd8-4fef-add2-07b7bd5307e3"
        src="music_note-24px.svg"
      />
       Name: 
      music_note-24px.svg
    </li>
  </div>
  <div>
    <li
      key="1b5ca3d3-485a-4da6-a184-0a8c792b2d95"
    >
       
      <img
        alt="remove_red_eye-24px.svg"
        key="bc27033f-735f-4d9f-b680-a42a67ce3f6e"
        src="remove_red_eye-24px.svg"
      />
       Name: 
      remove_red_eye-24px.svg
    </li>
  </div>
  <div>
    <li
      key="7b0e9d82-4780-429c-82cc-5641746fad64"
    >
       
      <img
        alt="security-24px.svg"
        key="41f4dadf-2ea7-46ba-97f3-7d2b4b363f8f"
        src="security-24px.svg"
      />
       Name: 
      security-24px.svg
    </li>
  </div>
  <div>
    <li
      key="a5b570a0-34e3-4a4f-8ea4-9b697465603f"
    >
       
      <img
        alt="straighten-24px.svg"
        key="2ef0a06e-f167-481c-a115-71a4d4cd2ee5"
        src="straighten-24px.svg"
      />
       Name: 
      straighten-24px.svg
    </li>
  </div>
  <div>
    <li
      key="00f3fe7e-9e32-4f05-b8b1-4e08ddbbe6b9"
    >
       
      <img
        alt="sync-24px.svg"
        key="2d699004-4977-4067-9f6d-184984191e0d"
        src="sync-24px.svg"
      />
       Name: 
      sync-24px.svg
    </li>
  </div>
  <div>
    <li
      key="b4bced17-9ca5-4fa0-acb3-1a1bb8c22785"
    >
       
      <img
        alt="toys-24px.svg"
        key="11429856-8f6a-455b-9ca3-06d956fab9eb"
        src="toys-24px.svg"
      />
       Name: 
      toys-24px.svg
    </li>
  </div>
  <div>
    <li
      key="a9c4e435-d0c4-4dd3-ab21-d10e8269bf2e"
    >
       
      <img
        alt="watch-24px.svg"
        key="9457dc69-1664-4233-b813-f88ef27935c6"
        src="watch-24px.svg"
      />
       Name: 
      watch-24px.svg
    </li>
  </div>
  <div>
    <li
      key="99ed74e8-5e99-43d2-9665-cf52bedfc009"
    >
       
      <img
        alt="agriculture-24px.svg"
        key="aa7f8c83-eea5-4309-a88a-0f6c75e0f4c3"
        src="agriculture-24px.svg"
      />
       Name: 
      agriculture-24px.svg
    </li>
  </div>
  <div>
    <li
      key="f0c10db8-53d8-4773-b186-440216159700"
    >
       
      <img
        alt="apartment-24px.svg"
        key="1c5119ca-bb99-4194-8f44-60e47f9e2a3e"
        src="apartment-24px.svg"
      />
       Name: 
      apartment-24px.svg
    </li>
  </div>
  <div>
    <li
      key="9d6b3642-fe06-41b4-b77d-ca96c77084a4"
    >
       
      <img
        alt="beach_access-24px.svg"
        key="229b1a9b-1d9e-4b4e-bad5-f2e752f29f2c"
        src="beach_access-24px.svg"
      />
       Name: 
      beach_access-24px.svg
    </li>
  </div>
  <div>
    <li
      key="e8582b02-517a-48d2-9fda-937f47abccbc"
    >
       
      <img
        alt="directions_bike-24px.svg"
        key="1ea612ad-6f5f-4336-ac0f-3c1348fc2bbe"
        src="directions_bike-24px.svg"
      />
       Name: 
      directions_bike-24px.svg
    </li>
  </div>
  <div>
    <li
      key="7f190972-2cfc-444a-bec4-d9e195a6074b"
    >
       
      <img
        alt="directions_boat-24px.svg"
        key="e97c32b5-5adf-488d-b691-4d4bf5ca2167"
        src="directions_boat-24px.svg"
      />
       Name: 
      directions_boat-24px.svg
    </li>
  </div>
  <div>
    <li
      key="7edb3e6b-0ae6-4434-a1fe-bcaee2efe044"
    >
       
      <img
        alt="directions_car-24px.svg"
        key="75ed97ab-5eda-4f44-9fe6-68ff16e50713"
        src="directions_car-24px.svg"
      />
       Name: 
      directions_car-24px.svg
    </li>
  </div>
  <div>
    <li
      key="2b9531a9-bc2c-4763-a75d-3f4c06a66031"
    >
       
      <img
        alt="flight-24px.svg"
        key="83d969f2-1aa6-44f4-9496-a5368d334df9"
        src="flight-24px.svg"
      />
       Name: 
      flight-24px.svg
    </li>
  </div>
  <div>
    <li
      key="f4f84b2c-35f8-4af2-8c71-f90a98371afe"
    >
       
      <img
        alt="home-24px.svg"
        key="bebfa390-d06c-460b-b4ff-4442935e89bb"
        src="home-24px.svg"
      />
       Name: 
      home-24px.svg
    </li>
  </div>
  <div>
    <li
      key="e302d61e-f6b4-43ce-afc6-15c08a14d1b3"
    >
       
      <img
        alt="hotel-24px.svg"
        key="091c292c-d043-4ad8-8db7-bc2b6b1c17ea"
        src="hotel-24px.svg"
      />
       Name: 
      hotel-24px.svg
    </li>
  </div>
  <div>
    <li
      key="57e0d285-f139-4cb7-a102-0cc1bd6b738f"
    >
       
      <img
        alt="insert_photo-24px.svg"
        key="6255ebc5-8f5e-4408-bd42-c2d75530bf34"
        src="insert_photo-24px.svg"
      />
       Name: 
      insert_photo-24px.svg
    </li>
  </div>
  <div>
    <li
      key="ed9955bd-bfa4-49f0-9d69-f446c5071523"
    >
       
      <img
        alt="local_cafe-24px.svg"
        key="7633d592-80e5-48b7-80e5-c5aeb4af5977"
        src="local_cafe-24px.svg"
      />
       Name: 
      local_cafe-24px.svg
    </li>
  </div>
  <div>
    <li
      key="7720dda9-2f97-4227-ac84-ffb32a1233ff"
    >
       
      <img
        alt="local_florist-24px.svg"
        key="40576c18-f5f4-4f89-949b-43c06b4ebdb6"
        src="local_florist-24px.svg"
      />
       Name: 
      local_florist-24px.svg
    </li>
  </div>
  <div>
    <li
      key="0d8ba149-7b5a-4b5e-9e36-1805def79fec"
    >
       
      <img
        alt="local_shipping-24px.svg"
        key="2ec2231c-5a43-41b7-8a65-dede5ed71102"
        src="local_shipping-24px.svg"
      />
       Name: 
      local_shipping-24px.svg
    </li>
  </div>
  <div>
    <li
      key="613c7f75-1816-4329-ab34-912e8adec5e2"
    >
       
      <img
        alt="luggage-24px.svg"
        key="e305a8ff-2b37-49d1-b73d-4973ef30053a"
        src="luggage-24px.svg"
      />
       Name: 
      luggage-24px.svg
    </li>
  </div>
  <div>
    <li
      key="24356e0a-8b41-49d8-95c7-4cfa9de4ed93"
    >
       
      <img
        alt="nature_people-24px.svg"
        key="f55032ae-55ee-4f0b-8d25-7c85ff6aed62"
        src="nature_people-24px.svg"
      />
       Name: 
      nature_people-24px.svg
    </li>
  </div>
  <div>
    <li
      key="f3d981e0-b6e6-4ec0-b1a9-8ec27389442b"
    >
       
      <img
        alt="pool-24px.svg"
        key="7f543a59-afd0-477e-97f5-2dd454456011"
        src="pool-24px.svg"
      />
       Name: 
      pool-24px.svg
    </li>
  </div>
  <div>
    <li
      key="4ee72a43-7c37-4483-85b6-b9d79c2a6982"
    >
       
      <img
        alt="ac_unit-24px.svg"
        key="aff4207b-30c0-4cd6-9e9f-b786119d3576"
        src="ac_unit-24px.svg"
      />
       Name: 
      ac_unit-24px.svg
    </li>
  </div>
  <div>
    <li
      key="45f96047-d2ee-4359-9f95-fac47b22e511"
    >
       
      <img
        alt="bar_chart-24px.svg"
        key="d305f663-ffbf-40f2-bb7e-aff8b69ab582"
        src="bar_chart-24px.svg"
      />
       Name: 
      bar_chart-24px.svg
    </li>
  </div>
  <div>
    <li
      key="75232fb4-dedb-4f1d-b4da-a7309d46e1b3"
    >
       
      <img
        alt="bedtime-24px.svg"
        key="c8eb8bc0-4911-44fc-bb17-3914c8572dc2"
        src="bedtime-24px.svg"
      />
       Name: 
      bedtime-24px.svg
    </li>
  </div>
  <div>
    <li
      key="c7b90ad8-016b-4f30-aef0-6c4920676152"
    >
       
      <img
        alt="brightness_low-24px.svg"
        key="b87ad2d9-7a25-4946-88cb-271aca49f44c"
        src="brightness_low-24px.svg"
      />
       Name: 
      brightness_low-24px.svg
    </li>
  </div>
  <div>
    <li
      key="ed95c969-e4ab-4abf-992c-209d2b27502e"
    >
       
      <img
        alt="bubble_chart-24px.svg"
        key="10ead9c1-44c5-4e72-affe-2eff546b4d2d"
        src="bubble_chart-24px.svg"
      />
       Name: 
      bubble_chart-24px.svg
    </li>
  </div>
  <div>
    <li
      key="f44e11f1-f99f-48fb-9697-b8afc2a3b470"
    >
       
      <img
        alt="cake-24px.svg"
        key="567f36e4-ced9-4ed7-bc2a-9850ae4cf924"
        src="cake-24px.svg"
      />
       Name: 
      cake-24px.svg
    </li>
  </div>
  <div>
    <li
      key="1ee99efb-3993-425b-b537-a6e6fca72d02"
    >
       
      <img
        alt="cloud_queue-24px.svg"
        key="d2272684-8aa0-4f04-beb2-b2f752c85610"
        src="cloud_queue-24px.svg"
      />
       Name: 
      cloud_queue-24px.svg
    </li>
  </div>
  <div>
    <li
      key="00b6e0c6-f508-4883-849f-fc8b006025a8"
    >
       
      <img
        alt="groups-24px.svg"
        key="8972e931-2845-4945-b1eb-37c9612996e6"
        src="groups-24px.svg"
      />
       Name: 
      groups-24px.svg
    </li>
  </div>
  <div>
    <li
      key="5f784981-bc53-41a8-ac75-6f25cd14690a"
    >
       
      <img
        alt="healing-24px.svg"
        key="9395a1d9-94ff-4f8b-8917-9bc838321712"
        src="healing-24px.svg"
      />
       Name: 
      healing-24px.svg
    </li>
  </div>
  <div>
    <li
      key="05e5c416-b3e5-4d74-9803-8deae0e2133e"
    >
       
      <img
        alt="insert_emoticon-24px.svg"
        key="cd26d6ce-5a36-4926-a06d-e4d56e6c147f"
        src="insert_emoticon-24px.svg"
      />
       Name: 
      insert_emoticon-24px.svg
    </li>
  </div>
  <div>
    <li
      key="e41c200f-8934-40ab-834d-12a506947a1c"
    >
       
      <img
        alt="miscellaneous_services-24px.svg"
        key="1b9a24a3-48a0-43e1-9a93-11bc155f1f06"
        src="miscellaneous_services-24px.svg"
      />
       Name: 
      miscellaneous_services-24px.svg
    </li>
  </div>
  <div>
    <li
      key="cae55ac2-ad4e-43e5-b5cc-c399bdfcfedb"
    >
       
      <img
        alt="mood_bad-24px.svg"
        key="311c419f-f8e8-4621-9579-b07a8af2cde1"
        src="mood_bad-24px.svg"
      />
       Name: 
      mood_bad-24px.svg
    </li>
  </div>
  <div>
    <li
      key="85df8bfa-9589-4161-b808-3bc09c409d4f"
    >
       
      <img
        alt="network_check-24px.svg"
        key="dc35e108-9d3b-413d-bd2d-3c3d62a2b681"
        src="network_check-24px.svg"
      />
       Name: 
      network_check-24px.svg
    </li>
  </div>
  <div>
    <li
      key="d7814986-65a7-4722-97f6-d9563722855a"
    >
       
      <img
        alt="self_improvement-24px.svg"
        key="b8fd7035-0aec-405f-bccc-75ec8dea9ad3"
        src="self_improvement-24px.svg"
      />
       Name: 
      self_improvement-24px.svg
    </li>
  </div>
  <div>
    <li
      key="01fa5fee-bb8a-411c-9eb7-bbc6c51324d8"
    >
       
      <img
        alt="star_border-24px.svg"
        key="c2e74319-7790-41d9-bd3d-963f6fb4dcb3"
        src="star_border-24px.svg"
      />
       Name: 
      star_border-24px.svg
    </li>
  </div>
  <div>
    <li
      key="e0ae9a2b-5baf-4ded-8fb3-b7dbf413bb2a"
    >
       
      <img
        alt="timer-24px.svg"
        key="24fe6bfd-3c60-4c83-8399-836d11b629a6"
        src="timer-24px.svg"
      />
       Name: 
      timer-24px.svg
    </li>
  </div>
</div>
`;
